docs/design/keyboard-shortcuts.md

```diff
@@ -256,6 +256,69 @@ When using custom keyboard shortcuts on the web, some keyboard shortcuts that ar
+> A good way to try out preview features is by using a Microsoft 365 subscription. If you don't already have a Microsoft 365 subscription, you can get one by joining the [Microsoft 365 developer program](https://developer.microsoft.com/office/dev-program).
+
+> [!NOTE]
+> The APIs described in this section require the [KeyboardShortcuts 1.1](../reference/requirement-sets/keyboard-shortcuts-requirement-sets.md) requirement set.
+
+To enable a user to customize the keyboard shortcuts, your add-in will need to provide a UI in which the user can assign new key combinations to the add-in's actions. The UI should present the user with the available actions and, for each, provide an input method, such as a text box, for the user to enter a key combination string. The UI should validate that the strings are the proper form. For example, there can be no spaces and keys must be separated by the "+" symbol. If the user is logged into Office, the custom combinations are saved in the user's roaming settings. If the user is not logged in, the customizations last only for the current session of the add-in.
+
+Use the [Office.actions.replaceShortcuts](/javascript/api/office/office.actions#replaceShortcuts) method to assign a user's custom keyboard combinations to your add-ins actions. The method takes a parameter of type `{[actionId:string]: string}`, where the `actionId`s are a subset of the action IDs that are defined in the add-in's extended manifest JSON. The values are the user's preferred key combinations. The following is an example in which some hypothetical custom key combinations are hardcoded. In an actual add-in, you create a function that gets them from the UI.
+
+```javascript
+const userCustomShortcuts = {
+    SHOWTASKPANE:"CTRL+SHIFT+1", 
+    HIDETASKPANE:"CTRL+SHIFT+2"
+};
+Office.actions.replaceShortcuts(userCustomShortcuts)
+    .then(function () {
+        console.log("Successfully registered.");
+    })
+    .catch(function (ex) {
+        if (ex.code == "ActionNotFound") {
+            console.log("ActionId does not exist.");
+        } else if (ex.code == "InvalidShortcut") {
+            console.log("Shortcut combination is invalid.");
+        }
+    });
+```
+
+To populate a UI that specifies the add-in's shortcuts that the user can customize, call the [Office.actions.getShortcuts](/javascript/api/office/office.actions#getShortcuts) method. This method returns an object of type `[actionId:string]:string|null}`, where the `actionId`s are (1) all of the action IDs that are defined in the add-in's extended manifest JSON and (2) all the customized shortcuts registered for the user in the user's roaming settings. The values are the key combinations currently assigned to the actions. The following is an example.
```

Comment:
Can we create a list here of the shortcuts that are returned with this API rather than listing them within the paragraph? For eg:
"The actionId shortcut pair returned by this method includes:
1. All of the actions that are defined in the add-in's extended manifest JSON AND
2. All of the customized actions registered for the user in the user's roaming settings
"